content/en/docs/concepts/extend-kubernetes/operator.md

```diff
@@ -6,10 +6,11 @@ weight: 30
 
 <!-- overview -->
 
-Operators are software extensions to Kubernetes that make use of
-[custom resources](/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
-to manage applications and their components. Operators follow
-Kubernetes principles, notably the [control loop](/docs/concepts/architecture/controller).
+Operators are software extensions to Kubernetes.
+
+An Operator consists of a set of [custom resources](/docs/concepts/extend-kubernetes/api-extension/custom-resources/#custom-resources) and [custom controllers](/docs/concepts/extend-kubernetes/api-extension/custom-resources/#custom-controllers). 
```

Comment:
This definition is not accurate. The custom resources or custom resource definitions doesn't have to be managed using an operator.

Reply:
> The custom resources or custom resource definitions don't have to be managed using an operator.
Yes.
But we are talking about the composition of an operator here.

Here is my definition.
Operator = custom-resources + custom-controllers

custom resources without custom controllers, I just call them custom-resources.
custom controllers without custom resources that watch built-in resources, I just call them custom-controllers.

Reply:
I'd suggest we remove this definition. "Operator" is a design pattern. It can be used to manage core k8s resources as well. For example, I may have an operator watching the creation and deletion of Namespaces and take some actions accordingly.

Reply:
> It can be used to manage core k8s resources as well. 

Yes, and in this case, I'll call it `custom controller`, not an operator.

Let's see what others say.

Reply:
I agree with @tengqm that Operator might not be appropriate here. https://github.com/kubeflow/training-operator/issues/300#issuecomment-357319596 captures this well:

> It's only an Operator if it's got: controller pattern + API extension + single-app focus.

Reply:
@nikhita 

Thanks for the reference information.

So, what does the operator consists of?

As a Kubernetes user, I just wanted an answer to this question.
I think we need the definition.
I know it's not perfect because we have lots of perspectives.

We should give an answer to Kubernetes users, not the developer's perspective on official documents.

Reply:
Here is reference information to support my opinion.

https://speakerdeck.com/mhausenblas/extending-kubernetes-101?slide=25
https://speakerdeck.com/mhausenblas/extending-kubernetes-101?slide=27

Reply:
> So, what does the operator consist of?

I think this is answered from the link @nikhita shared above:

> controller pattern + API extension + single-app focus.

As a Kubernetes user / operator, it is my opinion that the biggest divergence here is the clear callout for the single-app focus and abstracting / generalizing "custom resources/definitions" as "API extensions."

If you're still interested in working on this, I would suggest editing it to encompass the above 2 points.